File: source/gui/GUIPopup.java
package gui;

import callback.EmptyCallback;
import imgui.ImGui;
import imgui.flag.ImGuiPopupFlags;

public final class GUIPopup {
    private GUIPopup() {}

    public static void close() {
        ImGui.closeCurrentPopup();
    }

    public static void windowPopup(String name, EmptyCallback callback) {
        if (ImGui.beginPopupContextWindow(name, ImGuiPopupFlags.MouseButtonRight | ImGuiPopupFlags.NoOpenOverItems)) {
            callback.method();
            ImGui.endPopup();
        }
    }

    public static void itemPopup(String name, EmptyCallback callback) {
        if (ImGui.beginPopupContextItem(name, ImGuiPopupFlags.MouseButtonRight)) {
            callback.method();
            ImGui.endPopup();
        }
    }
}

File: source/gui/helper/GUIPopup.kt
package gui.helper

import imgui.ImGui
import imgui.flag.ImGuiPopupFlags

object GUIPopup {
    fun close() {
        ImGui.closeCurrentPopup()
    }

    fun windowPopup(name: String?, callback: () -> Unit) {
        if (ImGui.beginPopupContextWindow(name, ImGuiPopupFlags.MouseButtonRight or ImGuiPopupFlags.NoOpenOverItems)) {
            callback()
            ImGui.endPopup()
        }
    }

    fun itemPopup(name: String?, callback: () -> Unit) {
        if (ImGui.beginPopupContextItem(name, ImGuiPopupFlags.MouseButtonRight)) {
            callback()
            ImGui.endPopup()
        }
    }
}